### PR TITLE
tests: skip update portions of the test if we don't have an update configured.

### PR DESCRIPTION
**Pull request by justinsb**
This may enable us to stop requiring SkipUpdate in future.

### CODE REVIEW

File: pkg/controller/dynamic/dynamic_controller_integration_test.go [hunk 1]
```diff
@@ -368,7 +372,7 @@ func testUpdate(ctx context.Context, t *testing.T, testContext testrunner.TestCo
 	// Update resource from test data
 	updateUnstruct := testContext.UpdateUnstruct.DeepCopy()
 	if updateUnstruct == nil {
```
Comment (maqiuyujoyce) [inline pkg/controller/dynamic/dynamic_controller_integration_test.go:374]: As we've added the check above, is this check still needed?

**Review by maqiuyujoyce — COMMENTED** [review]
/lgtm

I'm okay with not requiring update here and requiring test coverage via other approaches later.

File: pkg/controller/dynamic/dynamic_controller_integration_test.go (hunk 1 above)
Comment (justinsb) [reply pkg/controller/dynamic/dynamic_controller_integration_test.go]: Good point; removed it!

File: pkg/controller/dynamic/dynamic_controller_integration_test.go [hunk 2]
```diff
@@ -367,9 +371,6 @@ func testUpdate(ctx context.Context, t *testing.T, testContext testrunner.TestCo
 
 	// Update resource from test data
 	updateUnstruct := testContext.UpdateUnstruct.DeepCopy()
-	if updateUnstruct == nil {
-		t.Fatalf("updateUnstruct is nil for '%v'. should SkipUpdate be set to true in resourcefixture/contexts?", testContext.ResourceFixture.Name)
-	}
```
Comment (acpana) [inline pkg/controller/dynamic/dynamic_controller_integration_test.go:372]: It seems like there's a plan to move away from `SkipUpdate` so disregard if this point will become moot in the future but as I onboarded on the code base, what I liked about this failure is that it gave me a strong signal for how to set up the fixtures tests.

Comment (maqiuyujoyce) [reply pkg/controller/dynamic/dynamic_controller_integration_test.go:372]: We probably need a better way to ensure coverage. This change isn't really part of it.

Regarding testdata, if we do want to enforce the existence of update.yaml, we probably should do it via a unit test but not the integration test.

**Review by maqiuyujoyce — COMMENTED** [review]
/lgtm
/approve

As long as we have the future plan to ensure update coverage, I'm good with it.

**Pull request merged**